lib/text/ui_text_displayer.js

```diff
@@ -121,10 +125,51 @@ shaka.text.UITextDisplayer = class {
       const containsCue = cuesList.some(
           (cueInList) => shaka.text.Cue.equal(cueInList, cue));
       if (!containsCue) {
-        this.cues_.push(cue);
+        additionalCues.push(cue);
+      }
+    }
+
+    // But we need to check if there are nested duplicates that this
```

Comment:
Since we already have shaka.text.Cue.equal, why wouldn't you add this check to that method?  It is already recursive, BTW.  See lib/text/cue.js.

Reply:
The circumstances I found this was that we would the cues would not be equal but would overlap (some of the same `nestedCues`), and we would only want to discard the overlap not the whole queue.
It did not feel right to use `shaka.text.Cue.equal`.  But maybe there's some merit in a new method e.g. `shaka.text.Cue.findOverlaps` or something like that.

Reply:
Ah, okay!  Now I get it.

So a single root cue contains multiple nested cues that need to be deduped within that single root cue?

Reply:
not quite... more like this
<img width="557" alt="image" src="https://user-images.githubusercontent.com/37144605/201020020-f08905ed-a138-4524-9200-6cb95e58bb4b.png">

Reply:
Hrm.   And what are the time ranges for N & N+1 in this diagram?  Overlapping?  Identical?

Do N+1, B(N+1), and C(N+1) all have the same time range?

Reply:
N & N+1 because the both contain B have overlapping time ranges
<img width="756" alt="image" src="https://user-images.githubusercontent.com/37144605/201281129-277d6ef2-bec4-4c1a-bafc-325a9d90b70a.png">